### PR TITLE
Add popup(toast) 

### PR DESCRIPTION
**Pull request by NikitaDenysenko**

### CODE REVIEW

File: src/components/notice-wrapper/notice-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Snackbar from '@material-ui/core/Snackbar';
+import Slide from '@material-ui/core/Slide';
+
+function TransitionUp(props) {
+  return <Slide {...props} direction='up' />;
+}
+
+const SnackbarMessage = () => {
```
Comment (moran711) [inline src/components/notice-wrapper/notice-wrapper.js:10]: Why do we need this component if you use `react-toastify`?

File: src/routes/routes.js [hunk 2]
```diff
@@ -132,6 +133,7 @@ const Routes = () => {
                 exact
                 render={({ match }) => {
                   const { category, model } = match.params;
+                  console.log(categories);
```
Comment (moran711) [inline src/routes/routes.js:136]: Please remove this `console.log`

File: src/routes/routes.js [hunk 3]
```diff
@@ -66,6 +66,7 @@ const Routes = () => {
                 component={PaymentsAndShipping}
               />
               <Route path='/privacy-policy' exact component={PrivacyPolicy} />
+              <Route path='/cart' exact component={Cart} />
```
Comment (moran711) [inline src/routes/routes.js:69]: We don't need this component here.

File: src/routes/routes.js [hunk 4]
```diff
@@ -20,6 +19,7 @@ const PaymentsAndShipping = lazy(() =>
   import('../pages/payments-and-shipping')
 );
 const PrivacyPolicy = lazy(() => import('../pages/privacy-policy'));
+const Cart = lazy(() => import('../pages/cart'));
```
Comment (moran711) [inline src/routes/routes.js:22]: We don't need this component here.

File: src/pages/product-details/product-submit/product-submit.js [hunk 5]
```diff
@@ -8,22 +8,27 @@ import FavouriteBorderIcon from '@material-ui/icons/FavoriteBorder';
...
 
+import { TOAST_MESSAGE } from '../../../translations/toast.translations';
+
 const ProductSubmit = ({ setSizeIsNotSelectedError, sizes }) => {
   const styles = useStyles();
   const dispatch = useDispatch();
   const { language, productToSend, product, wishlistItems } = useSelector(
-    ({ Language, Products, User, Wishlist }) => ({
+    ({ Language, Products, User, Wishlist, Toast }) => ({
       language: Language.language,
```
Comment (moran711) [inline src/pages/product-details/product-submit/product-submit.js:32]: Please move this selector to separate file.

File: src/pages/product-details/product-submit/product-submit.js [hunk 6]
```diff
@@ -8,22 +8,27 @@ import FavouriteBorderIcon from '@material-ui/icons/FavoriteBorder';
...
 } from '../../../translations/product-details.translations';
 
+import { TOAST_MESSAGE } from '../../../translations/toast.translations';
+
 const ProductSubmit = ({ setSizeIsNotSelectedError, sizes }) => {
   const styles = useStyles();
   const dispatch = useDispatch();
   const { language, productToSend, product, wishlistItems } = useSelector(
-    ({ Language, Products, User, Wishlist }) => ({
+    ({ Language, Products, User, Wishlist, Toast }) => ({
```
Comment (moran711) [inline src/pages/product-details/product-submit/product-submit.js:31]: You don't need to add Toast if you didn't get data from it.

File: src/redux/toast/toast.sagas.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { takeEvery } from 'redux-saga/effects';
+
+import { toast } from 'react-toastify';
+
+import { SET_TOAST_MESSAGE } from './toast.types';
+
+export function* handleSetToastMessage({ payload: message }) {
```
Comment (moran711) [inline src/redux/toast/toast.sagas.js:7]: We don't need saga for this.

File: src/pages/product-details/product-submit/product-submit.js [hunk 8]
```diff
@@ -93,6 +103,7 @@ const ProductSubmit = ({ setSizeIsNotSelectedError, sizes }) => {
 
   return (
     <div className={styles.submit}>
+      <Toast autoClose={3000} />
```
Comment (moran711) [inline src/pages/product-details/product-submit/product-submit.js:106]: Why do you use this component, if you have `ToastContainer `?

File: src/components/toast/toast.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+const Toast = (props) => (
+  <div>
+    <ToastContainer {...props} />
```
Comment (moran711) [inline src/components/toast/toast.js:7]: Why do you set this component in `components `folder if you name it `ToastContainer`?

**Review by moran711 — COMMENTED** [review]
You can create your own toast component without using libraries.

File: src/components/notice-wrapper/notice-wrapper.js (hunk 1 above)
Comment (NikitaDenysenko) [reply src/components/notice-wrapper/notice-wrapper.js:10]: I thought i deleted it) . Now it is deleted

File: src/routes/routes.js (hunk 2 above)
Comment (NikitaDenysenko) [reply src/routes/routes.js:136]: it's done

File: src/routes/routes.js (hunk 3 above)
Comment (NikitaDenysenko) [reply src/routes/routes.js:69]: i removed it

File: src/routes/routes.js (hunk 4 above)
Comment (NikitaDenysenko) [reply src/routes/routes.js:22]: it's removed

File: src/pages/product-details/product-submit/product-submit.js (hunk 5 above)
Comment (NikitaDenysenko) [reply src/pages/product-details/product-submit/product-submit.js:32]: done

File: src/pages/product-details/product-submit/product-submit.js (hunk 6 above)
Comment (NikitaDenysenko) [reply src/pages/product-details/product-submit/product-submit.js:31]: i removed it

File: src/pages/product-details/product-submit/product-submit.js (hunk 5 above)
Comment (NikitaDenysenko) [reply src/pages/product-details/product-submit/product-submit.js:32]: done

File: src/components/toast/toast.js (hunk 9 above)
Comment (NikitaDenysenko) [reply src/components/toast/toast.js:7]: it's removed

File: src/pages/product-details/product-submit/product-submit.js (hunk 8 above)
Comment (NikitaDenysenko) [reply src/pages/product-details/product-submit/product-submit.js:106]: it is deleted

File: src/redux/toast/toast.sagas.js (hunk 7 above)
Comment (NikitaDenysenko) [reply src/redux/toast/toast.sagas.js:7]: The hole logic is build around redux and redux-saga. It is done, to make reusability of toast easier

File: src/routes/routes.js [hunk 10]
```diff
@@ -15,6 +17,8 @@ import ProductsTable from '../pages/products-table';
 import ProductDetails from '../pages/product-details';
 import AboutUs from '../pages/about-us';
 
+import 'react-toastify/dist/ReactToastify.css';
```
Comment (moran711) [inline src/routes/routes.js:20]: I think It's better to set this styles in `src/index.js`

File: src/routes/routes.js [hunk 11]
```diff
@@ -49,10 +53,11 @@ const Routes = () => {
 
   return (
     <ConnectedRouter history={history}>
-      <Suspense fallback={<Loader />}>
+      <Suspense fallback={<div />}>
         <ErrorBoundary>
           <AppHeader />
           <div className={styles.root}>
+            <ToastContainer autoClose={3000} hideProgressBar />
```
Comment (moran711) [inline src/routes/routes.js:60]: I think it's better to set this props from store.

File: src/routes/routes.js (hunk 10 above)
Comment (NikitaDenysenko) [reply src/routes/routes.js:20]: done

File: src/routes/routes.js (hunk 11 above)
Comment (NikitaDenysenko) [reply src/routes/routes.js]: done

File: src/containers/toast/toast.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import { ToastContainer } from 'react-toastify';
+
+import { useSelector } from 'react-redux';
+
+const Toast = () => {
+  const toastSettings = useSelector((state) => state.Toast.toastSettings);
```
Comment (moran711) [inline src/containers/toast/toast.js:8]: Please move this selector to separate file.

**Review by kapoliub — APPROVED** [review]

File: src/pages/product-details/product-submit/product-submit.js [hunk 13]
```diff
@@ -8,30 +8,37 @@ import FavouriteBorderIcon from '@material-ui/icons/FavoriteBorder';
...
-      language: Language.language,
-      productToSend: Products.productToSend,
-      product: Products.product,
-      userData: User.userData,
-      wishlistItems: Wishlist.list
-    })
+    selectLanguageProductsUserWishlist
   );
 
+  const toastSettings = {
```
Comment (moran711) [inline src/pages/product-details/product-submit/product-submit.js:37]: I think it's better to move this to separate file.

File: src/containers/toast/toast.js (hunk 12 above)
Comment (NikitaDenysenko) [reply src/containers/toast/toast.js:8]: done

File: src/pages/product-details/product-submit/product-submit.js (hunk 13 above)
Comment (NikitaDenysenko) [reply src/pages/product-details/product-submit/product-submit.js:37]: done

File: src/containers/toast/toast.js (hunk 12 above)
Comment (NikitaDenysenko) [reply src/containers/toast/toast.js:8]: done

**Review by moran711 — APPROVED** [review]

**Review by ivan-bonk — APPROVED** [review]

**Pull request merged**